R/call.R

```diff
@@ -21,8 +21,8 @@
 #'
 #' @param .fn Function to call. Must be a callable object: a string,
 #'   symbol, call, or a function.
-#' @param ... Arguments to the call either in or out of a list. These dots
-#'   support [tidy dots][tidy-dots] features. Empty arguments are preserved.
+#' @param ... [[Dynamic][dyn-dots]] - Arguments for the function
```

Comment:
I'd be tempted by something like:

```R
#' @param ... <[dynamic][dyn-dots]> Arguments for the function call...
```

Reply:
That does look nice as well:

![dyn-dots2](https://user-images.githubusercontent.com/4465050/67991882-282a1180-fc11-11e9-9fc5-786c6ff05c97.png)

I have a slight preference for the square brackets, which one do you prefer?

Reply:
I prefer `<dynamic>` because it looks more like a type specification (which it kind of is), and I think at the level of documentation reader trying to make the distinction more clear will result in more net confusion.

I'm imagining we'll eventually get to something like this:

```R
#' @param df <data.frame()>
#' @param x <in_mask, vector()>
#' @param flag <logical(1)> 
#' @param name <character(1)>
```

Where the goal is not to fully capture the exact type, but give hints to the educated user in a form that they can internalise.

Reply:
Maybe the way to think of it is not exactly as a type specification, but as a set of modifiers that imply something about how each argument is interpreted.

Reply:
OTOH since we write types as prototypes, and prototypes can be expressed as function calls, maybe we should consider the same approach for data docs? OTOOH maybe the lack of `()` would help to distinguish between types and other modifiers.

Reply:
I agree we don't need to make the distinction clear in general purpose documentation. However it's still a worthwhile distinction to maintain in general. "Dynamic" refers to a language effect and describes how values are produced. "Type" refers to the set of values accepted by the function. One is about syntax, the other about semantics. Together they describe the _kind_ of inputs accepted by a function (e.g. what kind of R code the user is allowed to type), which is why it makes sense to treat them similarly at a high level.

I was leaning towards `[` because it looks like a button. From an aesthetic point of view I also like having CamelCase prose for topic descriptions, instead of code-based snake_case. One issue with code is that it's not clear how to convey type parameters such as size, and it's not clear whether constructors are uniform enough to make sure they are easy to parse by pressed users. Maybe we should use type abbreviations instead?

Could something like this also work?

```r
@param ... [Dynamic, DataMasked, Recycled, Type:lgl, Size:nrow] Variables to filter by.
```

Reply:
With specifiers like `Recycled`, we not only have a description of the kind of accepted inputs, but also a description of input validation or processing. Coercion semantics would also be of this nature.

Reply:
Let's do a poll to decide this

Reply:
(After having considered what `nse-force` would look like in CamelCase I'm no longer attracted to that casing for topics...)

Reply:
Now using angular brackets:

```r
#> @param ... <[dynamic][dyn-dots]> A list of named attributes.
```